src/main/java/org/springframework/data/couchbase/core/query/Query.java

```diff
@@ -323,18 +323,24 @@ public String toN1qlRemoveString(ReactiveCouchbaseTemplate template, String coll
 	 * @return QueryOptions
 	 */
 	public QueryOptions buildQueryOptions(QueryScanConsistency scanConsistency) {
-		final QueryOptions options = QueryOptions.queryOptions();
+		QueryOptions options = QueryOptions.queryOptions();
+		if (options == null) { // add/override what we got from PseudoArgs
```

Comment:
that line does not make sense, since the line above you always set it to non-null?

Reply:
Right.  The code is copied from the Collections branch to reduce the future merge.   And in that code 'options' is an argument and needs to be checked.